### PR TITLE
Move version file reading to `uv-toolchain`

### PR DESCRIPTION
**Pull request by zanieb**
In preparation for using this in other commands.

### CODE REVIEW

File: crates/uv-toolchain/src/version_files.rs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+use std::{io, path::PathBuf};
+use tracing::debug;
+
+use crate::ToolchainRequest;
+
+/// Read [`ToolchainRequest`]s from a version file, if present.
+///
+/// Prefers `.python-versions` then `.python-version`.
+/// If only one Python version is desired, use [`request_from_version_files`] which prefers the `.python-version` file.
+pub async fn requests_from_version_files() -> Result<Option<Vec<ToolchainRequest>>, io::Error> {
```
Comment (konstin) [inline crates/uv-toolchain/src/version_files.rs:11]: Should those be methods on `ToolchainRequest`?

**Review by konstin — APPROVED** [review]

File: crates/uv-toolchain/src/version_files.rs (hunk 1 above)
Comment (zanieb) [reply crates/uv-toolchain/src/version_files.rs:11]: That's a good question, I guess they could be? It seems a little weird to me to return `Vec<Self>`?

Comment (BurntSushi) [reply crates/uv-toolchain/src/version_files.rs:11]: I could go either way personally.

Does this need to be `Option<Vec<T>>` though? Can we just use `Vec<T>`? I think the former is necessary if you specifically need to distinguish between "not specified" and "empty sequence."

**Review by BurntSushi — APPROVED** [review]

File: crates/uv-toolchain/src/version_files.rs (hunk 1 above)
Comment (zanieb) [reply crates/uv-toolchain/src/version_files.rs:11]: We want to distinguish between the file does not exist and the file exists but is empty.

Comment (zanieb) [reply crates/uv-toolchain/src/version_files.rs:11]: I'll save moving these methods for after the stack is merged (if it makes sense) to save myself some rebasing.

**Pull request merged**